documentation/new-version-guideline.md

```diff
@@ -0,0 +1,302 @@
+
+```go
+pager := rgClient.NewListPager(nil)
+var resourceGroups []*armresources.ResourceGroup
+for pager.More() {
+    nextResult, err := pager.NextPage(ctx)
+    if err != nil {
+        // handle error...
+    }
+    if nextResult.ResourceGroupListResult.Value != nil {
+        resourceGroups = append(resourceGroups, nextResult.ResourceGroupListResult.Value...)
+    }
+}
+// dealing with `resourceGroups`
+```
+
+> NOTE: No IO calls are made until the NextPage() method is invoked. The read consistency across pages is determined by the service implement.
+
+### Item iterator
+
+If you are not care about the underlaying detail about the pageable operation, you could use the following generic utility to create a per-item iterator for all pageable operation.
+
+***Item iterator utility***
+```go
+type PageConstraint[TItem any] interface {
+	Items() []*TItem
+}
+
+type Iterator[TItem any, TPage PageConstraint[TItem]] struct {
+	pager *runtime.Pager[TPage]
+	cur   []*TItem
+	index int
+}
+
+func (iter *Iterator[TItem, TPage]) More() bool {
+	return iter.pager.More() || iter.index < len(iter.cur)
+}
+
+func (iter *Iterator[TItem, TPage]) NextItem(ctx context.Context) (*TItem, error) {
+	if iter.index == len(iter.cur) && !iter.pager.More() {
+		return nil, errors.New("no more items")
+	}
+	if iter.cur == nil || iter.index == len(iter.cur) {
+		// first page or page exhausted
+		page, err := iter.pager.NextPage(ctx)
+		if err != nil {
+			return nil, err
+		}
+		iter.cur = page.Items()
+		iter.index = 0
+	}
+	item := iter.cur[iter.index]
+	// advance item
+	iter.index++
+	return item, nil
+}
+
+func NewIterator[TItem any, TPage PageConstraint[TItem]](pager *runtime.Pager[TPage]) *Iterator[TItem, TPage] {
+	return &Iterator[TItem, TPage]{
+		pager: pager,
+	}
+}
+```
+
+***Usage***
+```go
+iter := NewIterator[*armresources.ResourceGroup](rgClient.NewListPager(nil))
+for iter.More() {
+    rg, err := iter.NextItem(ctx)
+    if err != nil {
+        // handle error...
+    }
+    // dealing with `rg`
+}
+```
+
+### Reference
+
+For more information, you could refer to [design guidelines of Paging](https://azure.github.io/azure-sdk/golang_introduction.html#methods-returning-collections-paging) and [API reference of pager](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime#Pager).
+
+## Long-Running Operations
+
+### General usage
+
+Some operations can take a long time to complete. Azure introduces the long-running operations (LROs) to do such operations asynchronously. You need to use Begin* to start an LRO. It will return a poller that can used to keep polling for the result until LRO is done. For examples:
+
+```go
+ctx := context.TODO()
+poller, err := client.BeginCreate(ctx, "resource_identifier", "additonal_parameter", nil)
+if err != nil {
+    // handle error...
+}
+resp, err = poller.PollUntilDone(ctx, 5 * time.Second)
```

Comment:
Shall we use the new usage of this method? I believe we now accepts an options instead of a frequency?

Reply:
After the code gen update, I'll update the doc. Besides this option, poller and pager package also changed.